components/constellation/constellation.rs

```diff
@@ -1628,17 +1661,17 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
                 Some(source_id)
             }
             None => {
-                let root_frame_id = self.root_frame_id;
+                let root_browsing_context_id = self.root_browsing_context_id;
 
                 // Make sure no pending page would be overridden.
-                for frame_change in &self.pending_frames {
-                    if frame_change.frame_id == root_frame_id {
+                for change in &self.pending_changes {
+                    if change.browsing_context_id == root_browsing_context_id {
                         // id that sent load msg is being changed already; abort
                         return None;
                     }
                 }
 
-                if !self.pipeline_is_in_current_frame(source_id) {
+                if self.get_activity(source_id) == DocumentActivity::Inactive {
```

Comment:
The problem with doing that is if we then traverse so the browsing context is fully active again, then the change will have been lost.